File: tests/project_euler_tests.rs
mod dt_test_utils;
use dt_test_utils::dt_run_file;

fn run_problem(n: &str) -> String {
    let filename = format!("tests/project_euler/problem-{}.dt", n);
    let res = dt_run_file(&filename);

    eprintln!("=== STDOUT:\n{}\n===\n", res.stdout);
    eprintln!("=== STDERR:\n{}\n===", res.stderr);

    res.stdout.trim_end().to_string()
}

#[test]
pub fn problem_01() {
    assert_eq!("233168", run_problem("01"));
}

#[test]
#[ignore = "Broken by zig impl changes"]
pub fn problem_02a() {
    assert_eq!("4613732", run_problem("02a"));
}

#[test]
#[ignore = "Broken by zig impl changes"]
pub fn problem_02b() {
    assert_eq!("4613732", run_problem("02b"));
}

#[test]
#[ignore = "Broken by zig impl changes"]
pub fn problem_03() {
    assert_eq!("6857", run_problem("03"));
}

#[test]
#[ignore = "Broken by zig impl changes"]
pub fn problem_04() {
    assert_eq!("906609", run_problem("04"));
}

File: src/tests/project_euler_tests.zig
const std = @import("std");
const expectEqualStrings = std.testing.expectEqualStrings;

const dtRunFile = @import("dt_test_utils.zig").dtRunFile;

test "problem_01" {
    try expectEqualStrings("233168\n", (try dtRunFile("./src/tests/project_euler/problem-01.dt")).stdout);
}

// Broken by zig impl changes

// test "problem_02a" {
//     try expectEqualStrings("4613732\n", (try dtRunFile("./src/tests/project_euler/problem-02a.dt")).stdout);
// }

// test "problem_02b" {
//     try expectEqualStrings("4613732\n", (try dtRunFile("./src/tests/project_euler/problem-02b.dt")).stdout);
// }

// test "problem_03" {
//     try expectEqualStrings("6857\n", (try dtRunFile("./src/tests/project_euler/problem-03.dt")).stdout);
// }

// test "problem_04" {
//     try expectEqualStrings("906609\n", (try dtRunFile("./src/tests/project_euler/problem-04.dt")).stdout);
// }
